.github/workflows/on-push.yaml

```diff
@@ -0,0 +1,254 @@
+        uses: actions/download-artifact@v4
+        with:
+          name: nextest-archives
+          path: .
+      - name: Run Unit Tests (Node)
+        run: pnpm --recursive test
+      - name: Run Unit Tests (Rust)
+        run: cargo-nextest nextest --color always run --no-fail-fast --retries 2 --archive-file nextest-archive.tar.zst
+
+  # Runs the Rust integration tests for the project. It depends on the
+  # `build-tests` job to build the nextest test archives and upload them as
+  # artifacts, however we do not define an explicit dependency with `needs`.
+  # This is because in this job, we also need to get the integration environment
+  # running, which generally takes around a minute. So, we start this job
+  # immediately and once the environment is running we use the `wait-other-jobs`
+  # action to wait until the `build-tests` job is complete and the artifacts are
+  # available. This lets us start the environment while the tests are building,
+  # and then run the tests as soon as the artifacts are available.
+  integration-tests:
+    name: Run Integration Tests
+    runs-on: ubuntu-24.04
+    strategy:
+      matrix:
+        partition: [1, 2]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: "recursive"
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          cache-key: "rust-tests"
+      - uses: pnpm/action-setup@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: "pnpm"
+      - uses: arduino/setup-protoc@v3
+        with:
+          version: "25.3"
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - run: pip install boto3
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: nextest@0.9.82
+      - name: Install Package Dependencies
+        run: make install
+      - name: Start Integration Test Environment
+        run: make integration-env-up-ci
+      - name: Wait for Test Artifacts
+        uses: kachick/wait-other-jobs@v3.6.0
+        timeout-minutes: 5
+        with:
+          retry-method: 'equal_intervals'
+          wait-seconds-before-first-polling: 1
+          min-interval-seconds: 5
+          wait-list: |
+            [
+              {
+                "workflowFile": "on-push.yaml",
+                "jobName": "Build Test Artifacts",
+                "optional": false,
+                "startupGracePeriod": {
+                  "minutes": 5
+                }
+              }
+            ]
+      - name: Download Nextest Archives
+        uses: actions/download-artifact@v4
+        with:
+          name: nextest-archives
+          path: .
+      - name: Run Integration Tests (Rust)
+        run: cargo-nextest nextest --color always run --no-fail-fast --test-threads 1 --retries 2 --partition hash:${{ matrix.partition }}/2 --archive-file nextest-archive-serial.tar.zst
```

Comment:
Let's remove the retries. If we construct a flaky test, let's go with handling flaky tests very quickly by either removing or fixing them when they arise.

Reply:
What does `--partition hash:${{ matrix.partition }}/2` do, and why is `--archive-file nextest-archive-serial.tar.zst` necessary?

Reply:
This splits the runs into 2 separate job runs based on the job matrix `[1,2]` -- i.e. we get two deterministic buckets of integration tests, splitting them roughly 50/50, so instead of one job taking ~3m, we get two jobs taking ~1.5m running in parallel.

More information in the nextest archive docs linked in PR description.

Reply:
Retries removed in [6500f48](https://github.com/stacks-network/sbtc/pull/1238/commits/6500f487ecd5a526a213469e8de4db58c3d1bf81)